components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3583,4 +3594,133 @@ protected String getSearchQuery(String searchQuery) throws APIManagementExceptio
         return criteria;
     }
 
+    @Override
+    public String getWSDLDocument(String username, String tenantDomain, String resourceUrl,
+            Map environmentDetails, Map apiDetails) throws APIManagementException {
+
+        if (username == null) {
+            username = APIConstants.END_USER_ANONYMOUS;
+        }
+        if (tenantDomain == null) {
+            tenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        }
+
+        Map<String, Object> docResourceMap = APIUtil.getDocument(username, resourceUrl, tenantDomain);
+        String wsdlContent = "";
+        if (log.isDebugEnabled()) {
+            log.debug("WSDL document resource availability: " + docResourceMap.isEmpty());
+        }
+        if (!docResourceMap.isEmpty()) {
+            try {
+                ByteArrayOutputStream arrayOutputStream = new ByteArrayOutputStream();
+                IOUtils.copy((InputStream) docResourceMap.get("Data"), arrayOutputStream);
+                String apiName = (String) apiDetails.get(API_NAME);
+                String apiVersion = (String) apiDetails.get(API_VERSION);
+                String environmentName = (String) environmentDetails.get(ENVIRONMENT_NAME);
+                String environmentType = (String) environmentDetails.get(ENVIRONMENT_TYPE);
+                if (log.isDebugEnabled()) {
+                    log.debug("Published SOAP api gateway environment name: " + environmentName + " environment type: "
+                            + environmentType);
+                }
+                byte[] updatedWSDLContent = this.getUpdatedWSDLByEnvironment(resourceUrl,
+                        arrayOutputStream.toByteArray(), environmentName, environmentType, apiName, apiVersion);
+                wsdlContent = new String(updatedWSDLContent);
+            } catch (IOException e) {
+                handleException("Error occurred while copying wsdl content into byte array stream for resource: "
+                        + resourceUrl);
+            }
+        } else {
+            handleException("No wsdl resource found for resource path: " + resourceUrl);
+        }
+        JSONObject data = new JSONObject();
+        data.put("contentType", docResourceMap.get("contentType"));
+        data.put("name", docResourceMap.get("name"));
+        data.put("Data", wsdlContent);
+        if (log.isDebugEnabled()) {
+            log.debug("Updated wsdl content details for wsdl resource: " + docResourceMap.get("name") + " is " +
+                    data.toJSONString());
+        }
+        return data.toJSONString();
+    }
+
+    /**
+     * This method is used updated wsdl with the respective environment apis are published
```

Comment:
I think there is a typo here. Shouldn't it be "...........is used to get the updated wsdl............."?